_posts/2023-04-04-segment-replication.md

```diff
@@ -0,0 +1,241 @@
+categories:
+ - technical-post
+meta_keywords: 
+meta_description: 
+
+excerpt: We are excited to announce that segment replication---a new replication strategy introduced in OpenSearch 2.3---is generally available in version 2.7. Segment replication is an alternative to document replication, where documents are copied from primary shards to their replicas. With segment replication, index segments are transferred instead, so the heavy-duty indexing workload is done on the primary shard, significantly increasing index throughput and lowering compute costs. In this blog post, we will dive deep into the concept of segment replication, its advantages and shortcomings, and planned future enhancements.
+---
+
+<style>
+table{
+    border:2px solid #e6e6e6;
+    display: block;
+    max-width: -moz-fit-content;
+    max-width: fit-content;
+    margin: 0 auto;
+    overflow-x: auto;
+}
+
+th{
+    border:2px solid #e6e6e6;
+    padding: 5px;
+    text-align: center;
+}
+
+td{
+    border:1px solid #e6e6e6;
+    padding: 10px;
+    text-align: center;
+}
+</style>
+
+We are excited to announce that segment replication---a new replication strategy introduced in OpenSearch 2.3---is generally available in version 2.7. Segment replication is an alternative to document replication, where documents are copied from primary shards to their replicas. With segment replication, index segments are transferred instead, so the heavy-duty indexing workload is done on the primary shard, significantly increasing index throughput and lowering compute costs. In this blog post, we will dive deep into the concept of segment replication, its advantages and shortcomings, and planned future enhancements.
+
+## Document replication
+
+Until version 2.7, OpenSearch achieved durability and search scaling by replicating data, ingesting documents on the primary and replica nodes. This methodology is known as _document replication_. The advantage of document replication is the low refresh latency between the primary node and replica nodes. However, document replication consumes more CPU power because the workload is duplicated on all nodes.
+
+All operations that affect an OpenSearch index (for example, adding, updating, or removing documents) are routed to one of the index’s primary shards. The primary shard is responsible for validating the operation and subsequently executing it locally. Once the operation has been completed successfully on the primary shard, the operation is forwarded to each of its replica shards in parallel. Each replica shard executes the operation, duplicating the processing performed on the primary shard. When the operation has completed (either successfully or with a failure) on every replica and a response has been received by the primary shard, the primary shard responds to the client. The response includes information about replication success or failure on replica shards. Refer to the following diagram for the document replication process.
```

Comment:
Each replica shard executes the same operation

Reply:
Reworded